### PR TITLE
[MOB-1969] - RN Android - Using rejecters where applicable

### PR DESCRIPTION
**Pull request by Ayyanchira**
Using rejecters where promises are available. In future, even in void functions, we should use rejecters to indicate failures in request fetched from RN side and promise resolve success if void methods are called successfully.

### CODE REVIEW

File: android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java [hunk 1]
```diff
@@ -238,7 +249,7 @@ public void getAttributionInfo(Promise promise) {
         IterableLogger.printInfo();
         IterableAttributionInfo attributionInfo = IterableApi.getInstance().getAttributionInfo();
         if (attributionInfo == null) {
-            promise.resolve(null);
+            promise.reject("","No attribution info found");
```
Comment (vbabenkoru) [inline android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java:252]:
```suggestion
            promise.reject("", "No attribution info found");
```

File: android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java [hunk 2]
```diff
@@ -267,6 +278,7 @@ public void getLastPushPayload(Promise promise) {
             promise.resolve(Arguments.fromBundle(IterableApi.getInstance().getPayloadData()));
         } else {
             IterableLogger.d(TAG, "No payload data found");
+            promise.reject("","No payload data found");
```
Comment (vbabenkoru) [inline android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java:281]:
```suggestion
            promise.reject("", "No payload data found");
```

File: android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java [hunk 3]
```diff
@@ -345,6 +357,7 @@ public void getInAppMessages(Promise promise) {
             promise.resolve(Serialization.convertJsonToArray(inAppMessageJsonArray));
         } catch (JSONException e) {
             IterableLogger.e(TAG, e.getLocalizedMessage());
+            promise.reject("","Failed to fetch messages with error " + e.getLocalizedMessage());
```
Comment (vbabenkoru) [inline android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java:360]:
```suggestion
            promise.reject("", "Failed to fetch messages with error " + e.getLocalizedMessage());
```

File: android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java [hunk 4]
```diff
@@ -108,6 +108,11 @@ public void updateEmail(String email) {
 
     @ReactMethod
     public void getEmail(Promise promise) {
+        String email = RNIterableInternal.getEmail();
+        if (email == null){
+            promise.reject("", "email is null");
```
Comment (vbabenkoru) [inline android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java:113]: Is this the expected behavior for a promise? I think `null` is a normal return value if the user hasn't logged in yet. I'm not sure about the semantics of a `Promise` but I thought it should only return a 'rejected' state if some error occurred?

Comment (Ayyanchira) [reply android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java:113]: We have to come to a consensus for that. This way RN developers dont have to rely on null checks for getEmail to proceed with their operations. And its like guarantee that something is always there if its a success.

File: android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java [hunk 5]
```diff
@@ -108,6 +108,11 @@ public void updateEmail(String email) {
 
     @ReactMethod
     public void getEmail(Promise promise) {
+        String email = RNIterableInternal.getEmail();
+        if (email == null){
```
Comment (vbabenkoru) [inline android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java:112]:
```suggestion
        if (email == null) {
```

File: android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java (hunk 4 above)
Comment (vbabenkoru) [reply android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java:113]: @tapashmajumder What do you think?

Comment (Ayyanchira) [reply android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java]: Had a chat with Tapash. Null email case will go to promise.resolve.

File: android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java [hunk 6]
```diff
@@ -142,6 +142,11 @@ public void updateUser(ReadableMap dataFields, Boolean mergeNestedObjects) {
 
     @ReactMethod
     public void getUserId(Promise promise) {
+        String userId = RNIterableInternal.getEmail();
```
Comment (vbabenkoru) [inline android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java:145]: Doesn't the same apply to `getUserId` as `getEmail`?

Comment (Ayyanchira) [reply android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java]: Ah.. yes totally skipped it 😬.
Thanks Victor

File: android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java [hunk 7]
```diff
@@ -238,7 +239,7 @@ public void getAttributionInfo(Promise promise) {
         IterableLogger.printInfo();
         IterableAttributionInfo attributionInfo = IterableApi.getInstance().getAttributionInfo();
         if (attributionInfo == null) {
-            promise.resolve(null);
+            promise.reject("", "No attribution info found");
```
Comment (vbabenkoru) [inline android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java:242]: Hmm, I didn't pay attention to this earlier, it seems similar to email and userId.. The value is `null`, so it shouldn't be an error.

File: android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java [hunk 8]
```diff
@@ -267,6 +268,7 @@ public void getLastPushPayload(Promise promise) {
             promise.resolve(Arguments.fromBundle(IterableApi.getInstance().getPayloadData()));
         } else {
             IterableLogger.d(TAG, "No payload data found");
+            promise.reject("", "No payload data found");
```
Comment (vbabenkoru) [inline android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java:271]: Same as above.

**Pull request merged**